x/group/typesupport.go

```diff
@@ -0,0 +1,40 @@
+package group
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/group/errors"
+)
+
+func (ms Members) ValidateBasic() error {
+	index := make(map[string]struct{}, len(ms.Members))
+	for i := range ms.Members {
+		member := ms.Members[i]
+		if err := member.ValidateBasic(); err != nil {
+			return err
+		}
+		addr := member.Address
+		if _, exists := index[addr]; exists {
+			return sdkerrors.Wrapf(errors.ErrDuplicate, "address: %s", addr)
+		}
+		index[addr] = struct{}{}
+	}
+	return nil
+}
+
+type AccAddresses []sdk.AccAddress
```

Comment:
where is this used?

Reply:
in `MsgCreateProposal.ValidateBasic()` where `AccAddresses.ValidateBasic()` is used to verify that there are not duplicates in the list of proposers.